File: src/Data_Structures/Queue/queue.ts
import { LinkedList } from "../Linked_List/linked-list";

export class Queue<T> {
  private list: LinkedList<T>;

  constructor(...items: T[]) {
    this.list = new LinkedList();
    items.map((item) => this.list.push(item));
  }

  enqueue(...items: T[]) {
    items.map((item) => this.list.push(item));
    return this.list.length;
  }

  dequeue() {
    return this.list.shift()?.value;
  }

  peek() {
    return this.list.head?.value;
  }

  get length() {
    return this.list.length;
  }

  isEmpty() {
    return this.list.isEmpty();
  }

  toArray() {
    return this.list.toArray();
  }
}

File: src/Data_Structures/Queue/queue.py
from src.Data_Structures.Linked_List.linked_list import LinkedList


class Queue:
    def __init__(self, *items):
        self.list = LinkedList()
        for item in items:
            self.list.push(item)

    def enqueue(self, *items):
        for item in items:
            self.list.push(item)

        return self.list.length

    def dequeue(self):
        return getattr(self.list.shift(), 'value', None)

    def peek(self):
        return getattr(self.list.head, 'value', None)

    def get_length(self):
        return self.list.length

    def is_empty(self):
        return self.list.is_empty()

    def to_list(self):
        return self.list.to_list()
